dotnet/src/Agents/Core/ChatCompletionAgent.cs

```diff
@@ -136,7 +145,22 @@ protected override Task<AgentChannel> CreateChannelAsync(CancellationToken cance
         return Task.FromResult<AgentChannel>(channel);
     }
 
-    private ChatHistory SetupAgentChatHistory(IReadOnlyList<ChatMessageContent> history)
+    private (IChatCompletionService service, PromptExecutionSettings? executionSettings) GetChatCompletionService(Kernel kernel, KernelArguments? arguments)
+    {
+        // Need to provide a KernelFunction to the service selector as a container for the execution-settings.
+        KernelFunction nullPrompt = KernelFunctionFactory.CreateFromPrompt("placeholder", arguments?.ExecutionSettings?.Values);
+        (IChatCompletionService chatCompletionService, PromptExecutionSettings? executionSettings) =
+            kernel.ServiceSelector.SelectAIService<IChatCompletionService>(
+                kernel,
+                nullPrompt,
+                arguments ?? []);
+
+        return (chatCompletionService, executionSettings);
+    }
+
+    private ChatHistory SetupAgentChatHistory(
+        IReadOnlyList<ChatMessageContent> history,
+        CancellationToken cancellationToken)
```

Comment:
Is this cancellation token used?

Reply:
Good eye.  That's a reminant of the templating POC that I carved out.  Removing.